test/vinyl-luatest/update_optimize_test.lua

```diff
@@ -0,0 +1,405 @@
+local t = require('luatest')
+local g = t.group()
+local fiber = require('fiber')
+local log = require('log')
+
+local space
+local dump_count
+local new_stmt_count
+local old_stmt_count
+local index, index2, index3
+
+g.before_all = function()
+    pcall(log.cfg, {level = 6})
+    box.cfg {
+        listen            = os.getenv("LISTEN"),
+        memtx_memory      = 512 * 1024 * 1024,
+        memtx_max_tuple_size = 4 * 1024 * 1024,
+        vinyl_read_threads = 2,
+        vinyl_write_threads = 3,
+        vinyl_memory = 512 * 1024 * 1024,
+        vinyl_range_size = 1024 * 64,
+        vinyl_page_size = 1024,
+        vinyl_run_count_per_level = 1,
+        vinyl_run_size_ratio = 2,
+        vinyl_cache = 10240, -- 10kB
+        vinyl_max_tuple_size = 1024 * 1024 * 6,
+    }
+
+    require('console').listen(os.getenv('ADMIN'))
+
+    space = box.schema.space.create('test', { engine = 'vinyl' })
+end
+
+local function dumped_stmt_count_two_indices(index, index2)
+    return index:stat().disk.dump.output.rows
+           + index2:stat().disk.dump.output.rows
+end
+
+local function dumped_stmt_count(index, index2, index3)
```

Comment:
Nit: instead of these two functions you could add one that would take an array to indexes and sum their stats.